packages/fuels-core/src/types/wrappers/transaction.rs

```diff
@@ -245,32 +208,30 @@ impl Transaction for TransactionType {
             TransactionType::Create(tx) => tx.witnesses(),
         }
     }
+}
 
-    fn witnesses_mut(&mut self) -> &mut Vec<Witness> {
-        match self {
-            TransactionType::Script(tx) => tx.witnesses_mut(),
-            TransactionType::Create(tx) => tx.witnesses_mut(),
-        }
-    }
-
-    fn with_witnesses(self, witnesses: Vec<Witness>) -> Self {
-        match self {
-            TransactionType::Script(tx) => TransactionType::Script(tx.with_witnesses(witnesses)),
-            TransactionType::Create(tx) => TransactionType::Create(tx.with_witnesses(witnesses)),
-        }
-    }
+#[derive(Debug, Clone)]
+pub(crate) struct UnresolvedSignature {
+    owner: Address,
+    secret_key: SecretKey,
 }
 
 macro_rules! impl_tx_wrapper {
     ($wrapper: ident, $wrapped: ident) => {
         #[derive(Debug, Clone)]
         pub struct $wrapper {
-            pub tx: $wrapped,
+            pub(crate) tx: $wrapped,
+            pub(crate) unresolved_signatures: Vec<UnresolvedSignature>,
+            pub(crate) consensus_parameters: ConsensusParameters,
         }
 
         impl From<$wrapped> for $wrapper {
             fn from(tx: $wrapped) -> Self {
-                $wrapper { tx }
+                $wrapper {
+                    tx,
+                    unresolved_signatures: Default::default(),
+                    consensus_parameters: Default::default(),
```

Comment:
@hal3e and I concluded that this could potentially lead to cases where the users holds a tx with wrong params.
When we fetch txs from the node, we convert them to our wrapper type before exposing it to the user. A potential risk here is that the user will use the default parameters somewhere down the line.

Reply:
Awesome catch! @FuelLabs/sdk-rust what do you think we make a dedicated `from_fuel_tx` that accepts a `fuel_tx` and the `consensus parameters` ?